src/dbnode/persist/fs/seek_manager.go

```diff
@@ -237,6 +254,10 @@ func (m *seekerManager) Test(id ident.ID, shard uint32, start time.Time) (bool,
 	return seekersAndBloom.bloomFilter.Test(id.Bytes()), nil
 }
 
+// Borrow allows for exclusive interaction with a previously opened seeker.
+// When a seeker is  currently Borrowed and another goroutine tries to Borrow it,
+// a Lock with a waitgroup ensures that the previous goroutine finishes its
+// operations before access is granted to the next caller.
```

Comment:
Maybe a little too much implementation detail here.

"Borrow returns a "borrowed" seeker which the caller has exclusive access to until it is returned later."

Reply:
Changed it